Traceback (most recent call last):
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/micromamba/envs/agu24workshop/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
with pygmt.config(GMT_DATA_SERVER="NOAA"):
    grid = pygmt.datasets.load_earth_relief(resolution="01d")
grid
------------------

----- stderr -----
gmtread [NOTICE]: Remote data courtesy of GMT data server NOAA [http://noaa.generic-mapping-tools.org]
----- stderr -----
gmtread [NOTICE]: SRTM15 Earth Relief v2.6 at 1x1 arc degrees reduced by Gaussian Cartesian filtering (314.5 km fullwidth) [Tozer et al., 2019].
----- stderr -----
gmtread [NOTICE]:   -> Download grid file [112K]: earth_relief_01d_g.grd
----- stderr -----
gmtread [ERROR]: Libcurl Error: Failure when receiving data from the peer
----- stderr -----
gmtread [ERROR]: Unable to obtain remote file @earth_relief_01d_g.grd
----- stderr -----
gmtread [NOTICE]:   -> Download grid file [112K]: earth_relief_01d_g.grd
----- stderr -----
gmtread [ERROR]: Libcurl Error: Failure when receiving data from the peer
----- stderr -----
gmtread [ERROR]: Unable to obtain remote file @earth_relief_01d_g.grd
----- stderr -----
gmtread [ERROR]: File @earth_relief_01d_g.grd not found
----- stderr -----
[Session pygmt-session (3)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mGMTCLibError[0m                              Traceback (most recent call last)
Cell [0;32mIn[2], line 2[0m
[1;32m      1[0m [38;5;28;01mwith[39;00m pygmt[38;5;241m.[39mconfig(GMT_DATA_SERVER[38;5;241m=[39m[38;5;124m"[39m[38;5;124mNOAA[39m[38;5;124m"[39m):
[0;32m----> 2[0m     grid [38;5;241m=[39m [43mpygmt[49m[38;5;241;43m.[39;49m[43mdatasets[49m[38;5;241;43m.[39;49m[43mload_earth_relief[49m[43m([49m[43mresolution[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43m01d[39;49m[38;5;124;43m"[39;49m[43m)[49m
[1;32m      3[0m grid

File [0;32m~/micromamba/envs/agu24workshop/lib/python3.12/site-packages/pygmt/datasets/earth_relief.py:176[0m, in [0;36mload_earth_relief[0;34m(resolution, region, registration, data_source, use_srtm)[0m
[1;32m    174[0m     [38;5;28;01mcase[39;00m [38;5;124m"[39m[38;5;124mgebco[39m[38;5;124m"[39m [38;5;241m|[39m [38;5;124m"[39m[38;5;124mgebcosi[39m[38;5;124m"[39m:
[1;32m    175[0m         name [38;5;241m=[39m [38;5;124m"[39m[38;5;124mearth_gebco[39m[38;5;124m"[39m
[0;32m--> 176[0m grid [38;5;241m=[39m [43m_load_remote_dataset[49m[43m([49m
[1;32m    177[0m [43m    [49m[43mname[49m[38;5;241;43m=[39;49m[43mname[49m[43m,[49m
[1;32m    178[0m [43m    [49m[43mprefix[49m[38;5;241;43m=[39;49m[43mprefix[49m[43m,[49m
[1;32m    179[0m [43m    [49m[43mresolution[49m[38;5;241;43m=[39;49m[43mresolution[49m[43m,[49m
[1;32m    180[0m [43m    [49m[43mregion[49m[38;5;241;43m=[39;49m[43mregion[49m[43m,[49m
[1;32m    181[0m [43m    [49m[43mregistration[49m[38;5;241;43m=[39;49m[43mregistration[49m[43m,[49m
[1;32m    182[0m [43m[49m[43m)[49m
[1;32m    183[0m [38;5;28;01mreturn[39;00m grid

File [0;32m~/micromamba/envs/agu24workshop/lib/python3.12/site-packages/pygmt/helpers/decorators.py:773[0m, in [0;36mkwargs_to_strings.<locals>.converter.<locals>.new_module[0;34m(*args, **kwargs)[0m
[1;32m    770[0m             bound[38;5;241m.[39marguments[[38;5;124m"[39m[38;5;124mkwargs[39m[38;5;124m"[39m][arg] [38;5;241m=[39m newvalue
[1;32m    772[0m [38;5;66;03m# Execute the original function and return its output[39;00m
[0;32m--> 773[0m [38;5;28;01mreturn[39;00m [43mmodule_func[49m[43m([49m[38;5;241;43m*[39;49m[43mbound[49m[38;5;241;43m.[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mbound[49m[38;5;241;43m.[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/micromamba/envs/agu24workshop/lib/python3.12/site-packages/pygmt/datasets/load_remote_dataset.py:416[0m, in [0;36m_load_remote_dataset[0;34m(name, prefix, resolution, region, registration)[0m
[1;32m    414[0m [38;5;28;01mwith[39;00m Session() [38;5;28;01mas[39;00m lib:
[1;32m    415[0m     [38;5;28;01mwith[39;00m lib[38;5;241m.[39mvirtualfile_out(kind[38;5;241m=[39m[38;5;124m"[39m[38;5;124mgrid[39m[38;5;124m"[39m) [38;5;28;01mas[39;00m voutgrd:
[0;32m--> 416[0m         [43mlib[49m[38;5;241;43m.[39;49m[43mcall_module[49m[43m([49m
[1;32m    417[0m [43m            [49m[43mmodule[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mread[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m    418[0m [43m            [49m[43margs[49m[38;5;241;43m=[39;49m[43m[[49m[43mfname[49m[43m,[49m[43m [49m[43mvoutgrd[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[43mbuild_arg_list[49m[43m([49m[43mkwdict[49m[43m)[49m[43m][49m[43m,[49m
[1;32m    419[0m [43m        [49m[43m)[49m
[1;32m    420[0m         grid [38;5;241m=[39m lib[38;5;241m.[39mvirtualfile_to_raster(outgrid[38;5;241m=[39m[38;5;28;01mNone[39;00m, vfname[38;5;241m=[39mvoutgrd)
[1;32m    422[0m [38;5;66;03m# Full path to the grid if not tiled grids.[39;00m

File [0;32m~/micromamba/envs/agu24workshop/lib/python3.12/site-packages/pygmt/clib/session.py:656[0m, in [0;36mSession.call_module[0;34m(self, module, args)[0m
[1;32m    654[0m status [38;5;241m=[39m c_call_module([38;5;28mself[39m[38;5;241m.[39msession_pointer, module[38;5;241m.[39mencode(), mode, argv)
[1;32m    655[0m [38;5;28;01mif[39;00m status [38;5;241m!=[39m [38;5;241m0[39m:
[0;32m--> 656[0m     [38;5;28;01mraise[39;00m GMTCLibError(
[1;32m    657[0m         [38;5;124mf[39m[38;5;124m"[39m[38;5;124mModule [39m[38;5;124m'[39m[38;5;132;01m{[39;00mmodule[38;5;132;01m}[39;00m[38;5;124m'[39m[38;5;124m failed with status code [39m[38;5;132;01m{[39;00mstatus[38;5;132;01m}[39;00m[38;5;124m:[39m[38;5;130;01m\n[39;00m[38;5;132;01m{[39;00m[38;5;28mself[39m[38;5;241m.[39m_error_message[38;5;132;01m}[39;00m[38;5;124m"[39m
[1;32m    658[0m     )

[0;31mGMTCLibError[0m: Module 'read' failed with status code 16:
gmtread [ERROR]: Libcurl Error: Failure when receiving data from the peer
gmtread [ERROR]: Unable to obtain remote file @earth_relief_01d_g.grd
gmtread [ERROR]: Libcurl Error: Failure when receiving data from the peer
gmtread [ERROR]: Unable to obtain remote file @earth_relief_01d_g.grd
gmtread [ERROR]: File @earth_relief_01d_g.grd not found

